ibis/backends/base/sql/compiler/select_builder.py

```diff
@@ -39,7 +39,9 @@ def get_result(self):
     def visit(self, node, in_subquery):
         in_subquery |= self.is_subquery(node)
 
-        for arg in node.args:
+        args = node if isinstance(node, ops.NodeList) else node.args
```

Comment:
`Annotable.__argnames__` contains the attribute names.

`Concrete.__args__` contains the attribute values.
`Concrete.args` is a backward compatible alias for `__args__`.
`Concrete.argnames` is a backward compatible for `__argnames__`.

Since `NodeList.values` is a tuple of nodes, `NodeList.args == ((values...),)` contains another level of nesting. This is where `__children__` comes into the picture since `Signature.unbind()` unflattens variadic arguments. 

In the future we could port the SQL compiler to use the graph traversal functions which would make this line obsolete.

Reply:
Got it, sounds good. We'll get to that port eventually!